apps/prairielearn/src/middlewares/authzCourseOrInstance.ts

```diff
@@ -205,420 +514,275 @@ export async function authzCourseOrInstance(req: Request, res: Response) {
-    // own course instances when they view them.
-    if (
-      idsEqual(user.user_id, res.locals.authn_user.user_id) &&
-      !res.locals.authz_data.has_course_instance_permission_view &&
-      !res.locals.authz_data.has_course_permission_view
-    ) {
-      res.locals.authz_data.has_student_access_with_enrollment = true;
-    }
-  }
+  res.locals.req_date = req_date;
 
-  res.locals.authz_data.overrides = overrides;
+  res.locals.authz_data = {
+    // Authn user data
+    authn_user: authnAuthResult.user,
+    authn_mode: authnAuthResult.mode,
+    authn_mode_reason: authnAuthResult.mode_reason,
+    authn_is_administrator: res.locals.is_administrator,
+    authn_course_role: authnAuthResult.course_role,
+    authn_has_course_permission_preview: authnAuthResult.has_course_permission_preview,
+    authn_has_course_permission_view: authnAuthResult.has_course_permission_view,
+    authn_has_course_permission_edit: authnAuthResult.has_course_permission_edit,
+    authn_has_course_permission_own: authnAuthResult.has_course_permission_own,
+    ...run(() => {
+      if (!req.params.course_instance_id) {
+        return {};
+      }
+      return {
+        authn_course_instance_role: authnAuthResult.course_instance_role,
+        authn_has_student_access: authnAuthResult.has_student_access,
+        authn_has_student_access_with_enrollment:
+          authnAuthResult.has_student_access_with_enrollment,
+        authn_has_course_instance_permission_view:
+          authnAuthResult.has_course_instance_permission_view,
+        authn_has_course_instance_permission_edit:
+          authnAuthResult.has_course_instance_permission_edit,
+      };
+    }),
+    // Effective user data
+    user: effectiveAuthResult.user,
+    mode: effectiveAuthResult.mode,
+    mode_reason: effectiveAuthResult.mode_reason,
+    is_administrator: effectiveUserData?.is_administrator ?? res.locals.is_administrator,
+    course_role: effectiveAuthResult.course_role,
+    has_course_permission_preview: effectiveAuthResult.has_course_permission_preview,
+    has_course_permission_view: effectiveAuthResult.has_course_permission_view,
+    has_course_permission_edit: effectiveAuthResult.has_course_permission_edit,
+    has_course_permission_own: effectiveAuthResult.has_course_permission_own,
+    ...run(() => {
+      if (!req.params.course_instance_id) {
+        return {};
+      }
+      return {
+        course_instance_role: effectiveAuthResult.course_instance_role,
+        has_student_access: effectiveAuthResult.has_student_access,
+        has_student_access_with_enrollment: effectiveAuthResult.has_student_access_with_enrollment,
+        has_course_instance_permission_view:
+          effectiveAuthResult.has_course_instance_permission_view,
+        has_course_instance_permission_edit:
+          effectiveAuthResult.has_course_instance_permission_edit,
+        user_with_requested_uid_has_instructor_access_to_course_instance:
```

Comment:
I would appreciate a sanity check of `user_with_requested_uid_has_instructor_access_to_course_instance`

Reply:
Added a comment in 34aa61a826142203e468da77ab516ce25d525421. Yes, there is a reason.